core/src/main/java/com/novoda/merlin/receiver/ConnectivityReceiver.java

```diff
@@ -24,6 +25,14 @@ private boolean connectivityAction(Intent intent) {
         return ConnectivityManager.CONNECTIVITY_ACTION.equals(intent.getAction());
     }
 
+    private boolean getIsConnected(Context context, Intent intent) {
+        if (intent.hasExtra(ConnectivityManager.EXTRA_NO_CONNECTIVITY)) {
+            return !intent.getBooleanExtra(ConnectivityManager.EXTRA_NO_CONNECTIVITY, false);
```

Comment:
yeah it's getting simpler, but *technically* this line of code is just a boolean check - where as the beard connects to a system service so is slower.

If we want to remove it we should also benchmark, lets leave that for another PR